### PR TITLE
Homework 6 - jwt & cors

### PR DESCRIPTION
**Pull request by JieLiuRiver**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -17,15 +18,16 @@ process.on('unhandledRejection', (err) => {
...
-const PORT = process.env.PORT || 3000;
+const PORT = process.env.PORT || config.port;
 const app = express();
 
 app.use(formatResponse);
 app.use(express.json());
+app.use(express.urlencoded());
 app.use(logApiMiddleware, logExceptionMiddleware);
 app.use(express.static('public'));
 app.use(cors());
```
Comment (JonCa1) [inline src/app.ts]: options for cors should be given, like cors origin, allowedHeaders, credentials, methods

File: src/config/index.ts [hunk 2]
```diff
@@ -7,5 +7,10 @@ export default {
         database: 'fhpeduef',
         port: 5432
     },
-    port: 3000
+    port: 3000,
+    api_base_url: '/api/v1',
+    jwt_key: '4a380a09-3aab-401b-a620-1372b7e8c77a',
```
Comment (JonCa1) [inline src/config/index.ts]: key/secret should be enabled to load from .env/.env.local... files

File: src/services/jwtServices.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import jwt from 'jsonwebtoken';
+import config from '@/config';
+import { JWTUserInfo, NextFunction, Request, Response } from '@/types';
+
+const verifyPromise = (token: string) => new Promise((resolve, reject) => {
```
Comment (JonCa1) [inline src/services/jwtServices.ts:5]: make this jwt token verification into a middleware

File: src/app.ts (hunk 1 above)
Comment (JieLiuRiver) [reply src/app.ts]: Ok, thanks for reminding

File: src/config/index.ts (hunk 2 above)
Comment (JieLiuRiver) [reply src/config/index.ts]: OK

**Pull request merged**